dev-libs/aws-sdk-cpp/aws-sdk-cpp-1.7.160.ebuild

```diff
@@ -0,0 +1,95 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit cmake-utils python-single-r1
+
+DESCRIPTION="AWS SDK for C++"
+HOMEPAGE="https://aws.amazon.com/sdk-for-cpp/"
+SRC_URI="https://github.com/aws/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+MODULES=(
+	access-management acm acm-pca alexaforbusiness amplify apigateway
```

Comment:
Seriously... I know this wasn't done by you, but you could make it better. Is it possible to pick few *important* flags, keep them and enable/disable **all** the others with a single USE flag (eg `extra-modules`?)
This would also make it super easy to check for required disk space and memory (bug #665540), since it'd be dependant on a single USE flag.

Reply:
Maybe some can be put together, like putting all kinesis modules together.

But generally this is a very difficult task. There are no important flags but `core` which is enabled by default.

Example: Someone who produces Lambdas will most probably not need anything else, unless their lambdas want to talk to, let's say, S3 or something.

However, you are, of course, right. The list of modules has become extremely vast! And most of these names only mean something for people who actually have to work with them.

Some of the modules depend on each other, and the build system will enable them. But some sort of catalog to reduce the (unholy!) amount of flags right now would be great.

I'll see what I can map during the week-end.

Maybe there are some AWS specialists here who can pipe in and give their 2 cents?